### PR TITLE
Kustomize rbd-mirror directory

### PR DESCRIPTION
**Pull request by ELENAGER**
This PR addresses the following comment: https://github.com/RamenDR/ramen/pull/1564#discussion_r1765567518

### CODE REVIEW

**Review by nirs — COMMENTED** [review]
Arranging the resource for volume replication and volume group replication makes looks reasonable, but is wrong for the way these resources are used. We need one kustomization for the start hook, and another for the test hook.

The layout can be something like:

```console
├── README.md
├── start
├── start-data
│   ├── kustomization.yaml
│   ├── rbd-mirror-secret.yaml
│   ├── rbd-mirror.yaml
│   └── vrc-sample.yaml
├── test
└── test-data
    ├── kustomization.yaml
    ├── rbd-pvc.yaml
    └── vr-sample.yaml
```

Note that rbd-mirror-secret.yaml is a template and we cannot deploy it using kustomization, but we can reploy rbd-mirror.yaml and vrc-sample.yaml using the kustomization.

This generic structure can work for lot of other addons and make this part is easier to maintain.

File: test/addons/rbd-mirror/start [hunk 1]
```diff
@@ -107,11 +107,7 @@ def configure_rbd_mirroring(cluster, peer_info):
     )
 
     print(f"Apply rbd mirror to cluster '{cluster}'")
-    kubectl.apply(
-        "--filename=rbd-mirror.yaml",
-        "--namespace=rook-ceph",
-        context=cluster,
-    )
+    kubectl.apply("--kustomize=start-data", context=cluster)
```
Comment (nirs) [inline test/addons/rbd-mirror/start:110]: So much nicer like this.

File: test/addons/rbd-mirror/start [hunk 2]
```diff
@@ -224,20 +220,12 @@ def restart_rbd_mirror_daemon(cluster):
 
 def deploy_vrc_sample(cluster):
     print(f"Applying vrc sample in cluster '{cluster}'")
-    kubectl.apply(
-        "--filename=vrc-sample.yaml",
-        "--namespace=rook-ceph",
-        context=cluster,
-    )
+    kubectl.apply("--kustomize=start-data", context=cluster)
```
Comment (nirs) [inline test/addons/rbd-mirror/start:223]: This is not needed, you can delete now the entire function.

File: test/addons/rbd-mirror/start [hunk 3]
```diff
@@ -224,20 +220,12 @@ def restart_rbd_mirror_daemon(cluster):
...
 
 
 def deploy_vgrc_sample(cluster):
     print(f"Applying vgrc sample in cluster '{cluster}'")
-    kubectl.apply(
-        "--filename=vgrc-sample.yaml",
-        "--namespace=rook-ceph",
-        context=cluster,
-    )
+    kubectl.apply("--kustomize=start-data", context=cluster)
```
Comment (nirs) [inline test/addons/rbd-mirror/start:228]: Same - delete the function.

File: test/addons/rbd-mirror/test [hunk 4]
```diff
@@ -56,11 +56,7 @@ def rbd_mirror_image_status(cluster, image):
 
 def test_volume_replication(primary, secondary):
     print(f"Deploying pvc {PVC_NAME} in cluster '{primary}'")
```
Comment (nirs) [inline test/addons/rbd-mirror/test:58]: This log deploy all tests resources, you can update to to describe what is does. See addons/rook-ceph/test for example.

File: test/addons/rbd-mirror/test [hunk 5]
```diff
@@ -72,11 +68,7 @@ def test_volume_replication(primary, secondary):
     )
 
     print(f"Deploying vr vr-sample in cluster '{primary}'")
-    kubectl.apply(
-        "--filename=vr-sample.yaml",
-        "--namespace=rook-ceph",
-        context=primary,
-    )
+    kubectl.apply("--kustomize=test-data", context=primary)
```
Comment (nirs) [inline test/addons/rbd-mirror/test:71]: Not needed since you already applied test-data before.

**Review by nirs — COMMENTED** [review]
Nice!

Missing: delete the test-data kustomization at the end of the test, instead of deleting the resources manually.

File: test/addons/rbd-mirror/start (hunk 2 above)
Comment (ELENAGER) [reply test/addons/rbd-mirror/start]: Done

File: test/addons/rbd-mirror/start (hunk 3 above)
Comment (ELENAGER) [reply test/addons/rbd-mirror/start]: Done

File: test/addons/rbd-mirror/test (hunk 4 above)
Comment (ELENAGER) [reply test/addons/rbd-mirror/test]: Done

File: test/addons/rbd-mirror/test (hunk 5 above)
Comment (ELENAGER) [reply test/addons/rbd-mirror/test]: Done

**Comment by ELENAGER** [comment]
> Nice!
> 
> Missing: delete the test-data kustomization at the end of the test, instead of deleting the resources manually.

Done

**Review by nirs — APPROVED** [review]

**Pull request merged**